src/averageAges.js

```diff
@@ -55,7 +64,20 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothers = people.filter(person => person.sex === 'f'
+  && people.some(women => women.mother === person.name));
+
+  const children = onlyWithSon ? people.filter(person =>
+    mothers.some(mother => mother.name === person.mother)
+    && person.sex === 'm')
+    : people.filter(person => mothers.some(mother =>
+      mother.name === person.mother));
+
+  const ageDifference = children.reduce((a, child) =>
+    a + (child.born - (mothers.find(mother =>
+      mother.name === child.mother)).born), 0) / children.length;
```

Comment:
it would be nice to create a callback, something like `const findMother = (mother) => .... ` and pass it into the method `find` as a parameter, it would much easier to read